store/store.go

```diff
@@ -302,6 +302,61 @@ func (s *Store[H]) HasAt(_ context.Context, height uint64) bool {
 	return height != uint64(0) && s.Height() >= height
 }
 
+// DeleteRange implements [header.Store] interface.
+func (s *Store[H]) DeleteRange(ctx context.Context, from, to uint64) error {
+	if from >= to {
+		return fmt.Errorf("header/store: invalid range(%d,%d)", from, to)
+	}
+
+	batch, err := s.ds.Batch(ctx)
+	if err != nil {
+		return fmt.Errorf("header/store: delete range batch: %w", err)
+	}
+
+	if err := s.deleteRange(ctx, batch, from, to); err != nil {
+		return fmt.Errorf("header/store: delete range: %w", err)
+	}
+
+	if err := batch.Commit(ctx); err != nil {
+		return fmt.Errorf("header/store: delete range commit: %w", err)
+	}
+	return nil
+}
+
+func (s *Store[H]) deleteRange(ctx context.Context, batch datastore.Batch, from, to uint64) error {
+	hashes := make([]header.Hash, 0, to-from)
+	for h := from; h < to; h++ {
+		hash, err := s.heightIndex.HashByHeight(ctx, h)
```

Comment:
This is gonna be noticeable in traces for big ranges and another reason for #263. 
no actionable here